iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/task/InnerSpaceCompactionTask.java

```diff
@@ -196,8 +197,27 @@ protected void setTargetFileForRecover(List<TsFileResource> resources) {
     }
   }
 
+  protected void prepareCompactionModFiles() throws IOException {
+    if (!TsFileResource.useSharedModFile) {
+      return;
+    }
+    TsFileResource firstSource = filesView.sourceFilesInLog.get(0);
+    TsFileResource firstTarget = filesView.targetFilesInPerformer.get(0);
+    ModFileManagement modFileManagement = firstSource.getModFileManagement();
+    ModificationFile modificationFile = modFileManagement.allocateFor(firstTarget);
```

Comment:
If a shared mods file has been created here and the resource file has not yet been serialized, can it be found during restart recovery?

Reply:
I think if the resource file is not serialized, then the compaction is not completed, and the target file will be removed eventually. 
So, it does not matter whether the shared mod file can be found in the recovery.

Reply:
If the shared mod file can not be found in the recovery and the compaction task will rollback in the system recovery, at this point, the compaction recover task may not be able to locate this shared mods file.

Reply:
Why should the recovery task locate the shared mod file during a roll-back?
If you are concerned about the deletion of the shared mod file, then it is okay.
Because after all TsFiles are recovered, all unreferenced shared mod files will be deleted (in another PR),
and it is fine for the compaction recovery task to leave the shared mod files alone.